azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/AbstractWebAppMojo.java

```diff
@@ -416,6 +398,30 @@ protected WebAppConfig getWebAppConfig() throws AzureExecutionException {
         return parser.parse();
     }
 
+    protected AzureAppService getOrCreateAzureAppServiceClient() throws AzureExecutionException {
+        try {
```

Comment:
protected Azure getOrCreateAzureClient() throws AzureAuthFailureException, AzureExecutionException {
        try {
            final MavenAuthConfiguration mavenAuthConfiguration = auth == null ? new MavenAuthConfiguration() : auth;
            mavenAuthConfiguration.setType(getAuthType());
            azureCredentialWrapper = MavenAuthUtils.login(session, settingsDecrypter, mavenAuthConfiguration, this.httpProxyHost, this.httpProxyPort);
            if (Objects.isNull(azureCredentialWrapper)) {
                return null;
            }

            final AzureEnvironment env = azureCredentialWrapper.getEnv();
            final String environmentName = AuthHelper.azureEnvironmentToString(env);
            if (env != AzureEnvironment.AZURE) {
                Log.prompt(String.format(USING_AZURE_ENVIRONMENT, TextUtils.cyan(environmentName)));
            }
            Log.info(azureCredentialWrapper.getCredentialDescription());
            final Proxy proxy = ProxyUtils.createHttpProxy(getHttpProxyHost(), Integer.toString(getHttpProxyPort()));
            final Azure tempAzure = Azure.configure()
                    .withProxy(proxy)
                    .authenticate(azureCredentialWrapper.getAzureTokenCredentials()).withDefaultSubscription();
            final PagedList<Subscription> subscriptions = tempAzure.subscriptions().list();
            subscriptions.loadAll();
            final String targetSubscriptionId = getTargetSubscriptionId(tempAzure, subscriptions);
            checkSubscription(subscriptions, targetSubscriptionId);
            azureCredentialWrapper.withDefaultSubscriptionId(targetSubscriptionId);
            return AzureClientFactory.getAzureClient(azureCredentialWrapper, getUserAgent(), getHttpProxyHost(), getHttpProxyPort());
        } catch (AzureLoginException | IOException e) {
            throw new AzureAuthFailureException(e.getMessage());
        }
    }